src/test/java/darkdata/web/AdvisoryControllerTest.java

```diff
@@ -0,0 +1,151 @@
+package darkdata.web;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import darkdata.DarkDataApplication;
+import darkdata.service.RecommendationService;
+import darkdata.web.api.RecommendationRequest;
+import darkdata.web.api.RecommendationResponse;
+import darkdata.web.api.candidate.CandidateWorkflow;
+import org.apache.commons.io.IOUtils;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.core.io.Resource;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.DoubleSummaryStatistics;
+import java.util.List;
+
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Matchers.anyListOf;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+
+
+/**
+ * @author anirudhprabhu
+ */
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = DarkDataApplication.class)
+@WebAppConfiguration("server.port=0")
+public class AdvisoryControllerTest {
```

Comment:
@anirudhprabhu  let's add 2 tests to this Test class.  A test that intentionally invokes a HTTP 400 response by sending a bad input to the service and a test that checks that the service returns a HTTP 500 if there is an internal error.